### PR TITLE
Add experimental Go ATS config

### PR DESCRIPTION
**Pull request by rob05c**
Adds experimental Go ATS config generator, as a library which can be used as a CLI app or HTTP service.

### CODE REVIEW

File: traffic_monitor/experimental/traffic_monitor/enum/enum.go [hunk 1]
```diff
@@ -44,6 +43,8 @@ type CacheType string
 const (
 	// CacheTypeEdge represents an edge cache.
 	CacheTypeEdge = CacheType("EDGE")
+	// CacheTypeEdgeVector represents a vector (teak) cache.
+	CacheTypeEdgeVector = CacheType("EDGE_VECTOR")
```
Comment (rob05c) [inline traffic_monitor/experimental/traffic_monitor/enum/enum.go]: This has to be removed before it can be merged, Vector is implementation-specific.

**Comment by knutsel** [comment]
@rob05c - should we still merge this? If so, can you add the Apache license?

**Comment by rob05c** [comment]
@knutsel Added. 

I'd still like to see it merged, yes. Separating ATS configuration into its own app is both a step toward microservices and away from monoliths, and a step toward decoupling Traffic Ops and our cache architecture.

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-traffic_ops-test-PR/19/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/34/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/627/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/584/

**Comment by dangogh** [comment]
I think this is a good idea,  but needs updating for changes to dependency paths:
```
go build cli/*
cli/ats_config_cli.go:19:2: cannot find package "github.com/apache/incubator-trafficcontrol/traffic_monitor_golang/traffic_monitor/trafficopswrapper" in any of:
...
```
I fixed this in my local clone,  but started cascading to more things that needed to be changed..

**Comment by dneuman64** [comment]
Is this something we still want in the project?  If so, then please fix the conflicts and find a committer to work with to get this committed.  If not, then please close the PR.

**Comment by dneuman64** [comment]
fair enough.  We can keep this open for now and re-evaluate next time I do a sweep of old PRs.

**Comment by rob05c** [comment]
Ok, the config generator hasn't been merged yet, but the code is all there, and it's a better reference than this:

https://github.com/apache/trafficcontrol/pull/3917

**Pull request closed without merging**